src/main/java/com/ryanharter/auto/value/moshi/AutoValueMoshiExtension.java

```diff
@@ -142,18 +150,39 @@ public String generateClass(Context context, String className, String classToExt
 
     Map<String, TypeName> types = convertPropertiesToTypes(context.properties());
 
+    List<? extends TypeParameterElement> typeParams = context.autoValueClass().getTypeParameters();
+    boolean shouldCreateGenerics = typeParams != null && typeParams.size() > 0;
+
     ClassName classNameClass = ClassName.get(context.packageName(), className);
     ClassName autoValueClass = ClassName.get(context.autoValueClass());
+    TypeName autoValueClassName = autoValueClass;
+    TypeVariableName[] genericTypeNames = null;
+
+    TypeName superclass;
 
-    TypeSpec typeAdapter = createTypeAdapter(classNameClass, autoValueClass, properties);
-    MethodSpec jsonAdapterMethod = createJsonAdapterMethod(autoValueClass, typeAdapter);
+    if(shouldCreateGenerics) {
+      genericTypeNames = new TypeVariableName[typeParams.size()];
+      for (int i = 0; i < typeParams.size(); i++)
+        genericTypeNames[i] = TypeVariableName.get(typeParams.get(i));
+
+      superclass = ParameterizedTypeName.get(ClassName.get(context.packageName(), classToExtend), (TypeName[]) genericTypeNames);
+      autoValueClassName = ParameterizedTypeName.get(autoValueClass, (TypeName[]) genericTypeNames);
+    }else {
+      superclass = TypeVariableName.get(classToExtend);
+    }
+
+    TypeSpec typeAdapter = createTypeAdapter(classNameClass, autoValueClassName, genericTypeNames, properties);
+    MethodSpec jsonAdapterMethod = createJsonAdapterMethod(autoValueClassName, genericTypeNames, typeAdapter);
 
     TypeSpec.Builder subclass = TypeSpec.classBuilder(className)
-        .superclass(TypeVariableName.get(classToExtend))
+        .superclass(superclass)
         .addType(typeAdapter)
         .addMethod(generateConstructor(types))
         .addMethod(jsonAdapterMethod);
 
+    if(shouldCreateGenerics)
+      subclass.addTypeVariables(Arrays.asList(genericTypeNames));
```

Comment:
ether make this a one liner or add curly brackets.
also missing space between `if` and bracket